yvalenta/search.rb

```diff
@@ -0,0 +1,55 @@
+class SearchVowels
+
+  def initialize
+    vowels
+  end
+
+  def open_file
+    f = File.open("../text.txt", "r")
+    run(f)
+    f.close
+  end
+
+  def time_start(start)
+    struct = Struct.new(:start)
+    @time_s = struct.new(start)
+  end
+
+  def time_finish(finish)
+    struct = Struct.new(:finish)
+    @time_f = struct.new(finish)
+  end
+
+  private
+
+  def vowels
+    @vowels ||= { 'a' => 0, 'e' => 0, 'i' => 0, 'o' => 0, 'u' => 0 }
+  end
+
+  def run file
+    time_start Time.now
+    file.each_line do |line|
+      line.downcase.split("").each do |c|
+        validate(c)
+      end
+    end
+    print(vowels)
+  end
+
+  def validate char
+    if !@vowels[char].nil?
+      @vowels[char] += 1
+    end
+    time_finish Time.now
+  end
+
+  def print vowels
```

Comment:
Is not necesary to do a loop for print the vowels, See that you have a method and a global variable, even is useless to receive an argument that you have as a global variable